### ISSUE TITLE
Additional files for a sourceport doesn't work

### ISSUE THREAD

**Issue by CortexReaver** [nstlaurent/DoomLauncher#255]

**Comment by nstlaurent** [comment]
Hi, can you provide some more information on this? I need to know what port and files you are using and what you are expecting vs what is happening. As much information as possible, thanks.

**Comment by CortexReaver** [comment]
The port I'm using is GZDoom, but Im don't know what I can provide. Uhhmmm, maybe this?
![30-05-2022 172356](https://user-images.githubusercontent.com/34350690/171012140-f1d16985-f8c2-4e6d-bde6-0df0137f6841.png)
As you can see, I have to write all .pk files through the "Parameters". Because it give me basically no result if I adding them in the list below.

**Comment by CortexReaver** [comment]
I figured it out... kinda. You have to reselect the sourceport, when launching some WAD. It adds "additional files" from the port list.
![30-05-2022 174950](https://user-images.githubusercontent.com/34350690/171016966-fd96e5be-dcc4-4a3d-9243-f4467f5c91cd.png)
Still, it is not intuitive. May be will it be better if adds files to the end of list without reselecting? And mark it somehow (with pot name of the end or some color), so user will know that it was added with sourceport options.

**Comment by nstlaurent** [comment]
This happens with files you have already launched. If download a new file and GZDoom is your default port, it will work as you would expect. This is actually not an easy problem to solve because Doom Launcher can't read peoples minds. It used to always add them on window load, but this created an issue where people sometimes didn't want the automatically included files and would remove them. Then they would have to remember to remove them every single launch and is very annoying.

I'm open to suggestions for this exact problem. Off the top of my head I could add a warning link next to the additional files saying some files may be missing, and clicking it will regenerate them. Will have to think about some better wording.

**Comment by CortexReaver** [comment]
![30-05-2022 174950](https://user-images.githubusercontent.com/34350690/171022458-0c21659b-c028-4107-bb82-b25ee0cff6fc.png)
Something like this, maybe. This is not final idea, just a sketch...

**Comment by CortexReaver** [comment]
Another mock-up, now the idea is to make Profiles global and not to tie to any particular mod.

So, you can add one more tab specially for profile management (or place it somewhere n options):
![31-05-2022 091325](https://user-images.githubusercontent.com/34350690/171112486-e5963152-52ba-47e9-9d2c-19e30b3bae83.png)

And then how it will be used "in a field":

![31-05-2022 094042](https://user-images.githubusercontent.com/34350690/171112852-80053a1f-fbcb-4bcd-a896-fc1a97081772.png)

"Plus" and "Minus" signs are for adding and removing entire profiles from the launch list respectively. Arrows - for moving them entirely. Source port and its profile are always interconnected, but showed by default as gray (inactive), unless you decide to click on the checkbox.

**Comment by snackerfork** [comment]
I agree with CortexWeaver's post. Right now the profile functionality seems barebones which is the only reason it's a pain to use (the otherwise fantastic) Doom Launcher. I'd like to be able to load from global, iWAD, and tag-based profiles so that I can quickly have a loadout between the same sets of mods but a different map/megawad, which would be especially helpful if using the "random" functionality.

![image](https://github.com/nstlaurent/DoomLauncher/assets/33107312/7ea14783-0bc7-48cd-a3da-82001993246a)

Here is an example of what I mean. If I want to play another map with this exact Corruption Cards loadout, I'd need to add 11 additional files. If you're using the random functionality especially, that's a huge pain. I think it'd be very useful if I could make the Corruption Cards profile a modifiable global profile so that I can select it and have the source port chosen and those files added automatically. Even better would be something like a "mods" tag which can have a number of mod "profiles" added to any map by checking them, so you could add, i.e., Corruption Cards, Gun Bonsai, and Final Doomer independently.

This is the best looking and most useful launcher to me, so I'd love to have this feature! Keep up the great work :)

**Comment by nstlaurent** [comment]
3.7.3 has the start of of global profile implementation to check out.
